### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by erlingrj**
1. Add assertions to all the tests to actually test things and catch with the CI
2. Move rodata and srodata to the .text section. I was facing some issues which were resolved by this. I don't fully understand why the previous configuration shouldn't work.

### CODE REVIEW

**Comment by erlingrj** [comment]
Actually, I am still facing problems with the linker script. Reordering the inputs in each section sometimes hides it. The problem is that some constant read-only data (in the .srodata section I believe) is not being initialized. After some debugging I am concluding that somehow the constant value is not present in the executable for some mysterious reason. 

These values are supposed to be copied from the program memory to the data memory at startup, but the data is not there in the program memory. Moving around the input sections a bit and suddenly the data appears in the program memory and can be copied into data memory. 

The error typically manifests itself in an illegal load operation which I often trace back to loading some global constant variable which turns out to be just zero.

Who actually wrote that linker script? I am a bit lost as to what causes this

File: programs/tests/c-tests/add/add.c [hunk 1]
```diff
@@ -15,6 +15,7 @@ int main() {
     uint32_t z = add(x, y);
     _fp_print(z);
 
+    assert(z == 3);
```
Comment (sberkun) [inline programs/tests/c-tests/add/add.c:18]: nit: any reason not to add asserts to the other two adds as well?

Comment (erlingrj) [reply programs/tests/c-tests/add/add.c:18]: There is only a single add statement here?

File: programs/lib/linker/flexpret.ld [hunk 2]
```diff
@@ -24,8 +24,8 @@ SECTIONS
     .data :
     {
         __data_start__ = .;
+        *(.rodata .rodata*)
```
Comment (sberkun) [inline programs/lib/linker/flexpret.ld:27]: @erlingrj  did we ever resolve why this works? Would it be possible to make a test that fails with the previous configuration?

File: programs/tests/c-tests/add/add.c (hunk 1 above)
Comment (sberkun) [reply programs/tests/c-tests/add/add.c:18]: oh sorry I misread; I thought the prints corresponding to x=1 and y=2 were also adds

**Review by sberkun — APPROVED** [review]
Everything looks good to me besides the linker script, which I'm unsure about. We could move that change into its own PR for further investigation.

**Comment by erlingrj** [comment]
I havent been able to reproduce the linker error it with simple tests. It showed up first when compiling LF programs... But I have reverted the linker script changes. Lets revisit this, I am creating an issue for it.

**Pull request merged**